src/unix/pipe.c

```diff
@@ -287,6 +287,58 @@ int uv_pipe_getpeername(const uv_pipe_t* handle, char* buffer, size_t* size) {
 }
 
 
+int uv_pipe_get_credentials(const uv_pipe_t* handle, uv_credentials_t* creds) {
+  int r;
+  uv_os_fd_t os_fd;
+
+  r = uv_fileno((uv_handle_t*) handle, &os_fd);
+  if (r != 0)
+    return r;
+
+#if defined(__APPLE__)
+  {
+    // https://stackoverflow.com/a/38207320
+    unsigned int uid;
+    unsigned int gid;
+    socklen_t pid_len = sizeof(creds->pid);
+
+    r = getsockopt(os_fd,
+                   SOCK_STREAM,
+                   LOCAL_PEERPID,
+                   &creds->pid,
+                   &pid_len);
+
+    if (r != 0)
+      return -errno;
+
+    r = getpeereid(os_fd, &uid, &gid);
+    if (r != 0)
+      return -errno;
+
+    creds->uid = (long) uid;
+    creds->gid = (long) gid;
+
+    return 0;
+  }
+#else
+  // https://stackoverflow.com/a/35827184/510036
+  struct ucred ucreds;
+  socklen_t len = sizeof(ucreds);
```

Comment:
Style (also applies to line 303): idiomatic libuv splits the definition and the assignment, i.e.:
```c
socklen_t len;

len = sizeof(ucreds);
r = getsocktopt(..., &len);
```

Reply:
:+1: